src/group/libp2p_ack_stream.erl

```diff
@@ -42,13 +42,23 @@ server(Connection, Path, _TID, Args) ->
     libp2p_framed_stream:server(?MODULE, Connection, [Path | Args]).
 
 init(server, Connection, [Path, AckModule, AckState | _]) ->
-    case AckModule:accept_stream(AckState, self(), Path) of
+    %% Catch errors from the handler module since the handling group
+    %% may have already stopped or crashed.
+    case (catch AckModule:accept_stream(AckState, self(), Path)) of
         {ok, AckRef} ->
             libp2p_connection:set_idle_timeout(Connection, ?ACK_STREAM_TIMEOUT),
             {ok, #state{connection=Connection,
                         ack_ref=AckRef, ack_module=AckModule, ack_state=AckState}};
         {error, Reason} ->
-            {stop, {error, Reason}}
+            {stop, {error, Reason}};
+        {'EXIT', {noproc, _}} ->
+            %% Handle noproc silently since it's pretty common for a
```

Comment:
shouldn't the workers be linked to or monitoring their group server?

Reply:
The workers and their streams are.. those are clients though.. This is a server clause. The stream handlers are registered at the swarm level. 

So either someone isn’t deregistering the stream handlers for a given path, or there’s a slew of them coming in before the de-registration is complete.